timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala

```diff
@@ -562,30 +562,38 @@ case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] wit
         )
     }
 
-    val absoluteLatency = lastSuccessTime.foldLeft(
+    val absoluteLatency = lastSuccessTime.map {
+      case (job, lastSuccess) => job -> getAbsoluteLatency(job, lastSuccess)
+    }.toMap
+
+    val latencies = List(
       Gauge(
         "cuttle_timeseries_scheduler_absolute_latency_epoch_seconds",
         "Absolute latency of a job in seconds, with respect to its last success with all previous executions being successful"
-      )
-    ) {
-      case (gauge, (job, lastSuccess)) =>
-        val expectedSuccess = job.scheduling.calendar match {
-          case Hourly => Instant.now.minus(1, HOURS)
-          case Daily(tz) => Instant.now.atZone(tz).minus(1, DAYS).toInstant
-          case Monthly(tz) => Instant.now.atZone(tz).minus(1, MONTHS).toInstant
+      ) -> absoluteLatency,
+      Gauge(
+        "cuttle_timeseries_scheduler_relative_latency_epoch_seconds",
+        "Relative latency of a job in seconds, taking into account its parents' latencies"
+      ) -> absoluteLatency.map {
+        case (job, latency) =>
+          val maxParentLatency = workflow.edges
+            .collect { case (v, dep, _) if v == job => dep }
+            .map(absoluteLatency.get(_)).flatten.foldLeft(0L)(Math.max)
+          job -> Math.max(0, latency - maxParentLatency)
+      }
+    ).map {
+      case (gauge, latencies) =>
```

Comment:
OPT: Could you remove the shadowing of latencies here by renaming the variable?

Reply:
Sure